pkg/jobrunaggregator/jobrunhistoricaldataanalyzer/analyzer.go

```diff
@@ -162,28 +138,54 @@ func (o *JobRunHistoricalDataAnalyzerOptions) compareAndUpdate(newData, currentD
 
 		newP99 := getDurationFromString(new.GetP99())
 		newP95 := getDurationFromString(new.GetP95())
+		newP75 := getDurationFromString(new.GetP75())
+		newP50 := getDurationFromString(new.GetP50())
 		d := parsedJobData{}
 
 		// If the current data contains the new data, check and record the time diff
 		if old, ok := currentData[key]; ok {
```

Comment:
The problem is likely here, for some reason, we never seem to have old data. Is it related to the value of key?